### PR TITLE
Upgrade Babel to 7.0.0 :tada:

### PR DESCRIPTION
**Pull request by Titozzz**
**Summary**

Thanks to babel team incredible work, they released babel 7 🎉 , here's an update to make sure we are not anymore on a beta version. This will enforce compatibily once every other package updates to babel 7

**Test plan**

yarn test is OK.

### CODE REVIEW

File: packages/metro-babel-register/package.json [hunk 1]
```diff
@@ -8,16 +8,16 @@
...
-    "@babel/plugin-proposal-class-properties": "7.0.0-beta.56",
-    "@babel/plugin-proposal-nullish-coalescing-operator": "7.0.0-beta.56",
-    "@babel/plugin-proposal-object-rest-spread": "7.0.0-beta.56",
-    "@babel/plugin-proposal-optional-catch-binding": "7.0.0-beta.56",
-    "@babel/plugin-proposal-optional-chaining": "7.0.0-beta.56",
-    "@babel/plugin-transform-async-to-generator": "7.0.0-beta.56",
-    "@babel/plugin-transform-flow-strip-types": "7.0.0-beta.56",
-    "@babel/plugin-transform-modules-commonjs": "7.0.0-beta.56",
-    "@babel/register": "7.0.0-beta.56",
+    "@babel/core": "7.0.0",
```
Comment (nicolo-ribaudo) [inline packages/metro-babel-register/package.json]: Since Babel 7 releases won't contain breaking change, you can/should use `^7.0.0`

Comment (Titozzz) [reply packages/metro-babel-register/package.json]: Oh yeah, sure, silly me 😣

**Comment by rafeca** [comment]
Yay!! 🎉 I'm also working on an internal commit to upgrade to `babel@7.0.0`.

Since at Facebook we use the master version of `Metro` we need to do some small tweaks to our internal babel plugins before merging this (there have been some breaking changes between `beta.56` and `7.0.0`), but this should be ready tomorrow 😊

**Comment by Titozzz** [comment]
Good to hear @rafeca ! Keep up the excellent work 😀

**Comment by Titozzz** [comment]
Hey @rafeca , I saw you merged Babel 7 into master, so I guess we can close this pull request ! Would it be possible to bump metro to a new version so that we can create a new RC for react native 0.57 ?

Thanks 🎉

**Pull request closed without merging**